python-package/lightgbm/sklearn.py

```diff
@@ -1169,6 +1215,19 @@ def feature_names_in_(self) -> np.ndarray:
 class LGBMRegressor(_LGBMRegressorBase, LGBMModel):
     """LightGBM regressor."""
 
+    def _more_tags(self) -> Dict[str, Any]:
+        # handle the case where ClassifierMixin possibly provides _more_tags()
+        if callable(getattr(_LGBMClassifierBase, "_more_tags", None)):
+            tags = _LGBMClassifierBase._more_tags(self)
```

Comment:
Proposing all these uses of `{some_class}.{some_method}` instead of `super().{some_method}` because we follow this advice from `scikit-learn`'s docs (https://scikit-learn.org/stable/developers/develop.html#rolling-your-own-estimator):

> *...mixins should be “on the left” while the `BaseEstimator` should be “on the right” in the inheritance list for proper MRO.*

Using `super()` would get the `_more_tags()` / `__sklearn_tags__()` from e.g. `sklearn.base.RegressorMixin`, but we want to use LightGBM's specific tags.